src/parser/parseUseDirectives.ts

```diff
@@ -1,79 +1,81 @@
 import { applyViewboxTransform } from './applyViewboxTransform';
 
-export function parseUseDirectives(doc) {
-  let nodelist = getMultipleNodes(doc, ['use', 'svg:use']),
-    i = 0;
-  while (nodelist.length && i < nodelist.length) {
-    const el = nodelist[i],
-      xlinkAttribute = el.getAttribute('xlink:href') || el.getAttribute('href');
+/**
+ * TODO: real docs, not sure what this is actually doing
+ * @param doc
+ */
+export function parseUseDirectives(doc: Document | HTMLElement) {
+  const nodeList = getMultipleNodes(doc, ['use', 'svg:use']);
+  for (const el of nodeList) {
+    const xlinkAttribute =
+      el.getAttribute('xlink:href') || el.getAttribute('href');
 
     if (xlinkAttribute === null) {
       return;
     }
 
-    var xlink = xlinkAttribute.slice(1),
-      x = el.getAttribute('x') || 0,
-      y = el.getAttribute('y') || 0,
-      el2 = elementById(doc, xlink).cloneNode(true),
+    const xlink = xlinkAttribute.slice(1),
+      x = el.getAttribute('x') ?? '0',
+      y = el.getAttribute('y') ?? '0',
+      namespace = svgNS;
+    let el2 = elementById(doc, xlink)?.cloneNode(true) as Element,
       currentTrans =
-        (el2.getAttribute('transform') || '') +
+        (el2?.getAttribute('transform') ?? '') +
         ' translate(' +
         x +
         ', ' +
         y +
         ')',
-      parentNode,
-      oldLength = nodelist.length,
-      attr,
-      j,
-      attrs,
-      len,
-      namespace = svgNS;
+      parentNode: ParentNode;
 
     applyViewboxTransform(el2);
     if (/^svg$/i.test(el2.nodeName)) {
       const el3 = el2.ownerDocument.createElementNS(namespace, 'g');
-      for (j = 0, attrs = el2.attributes, len = attrs.length; j < len; j++) {
-        attr = attrs.item(j);
-        el3.setAttributeNS(namespace, attr.nodeName, attr.nodeValue);
-      }
+      Array<Attr>
+        .from(el2.attributes)
+        .forEach((attr: Attr) =>
+          el3.setAttributeNS(namespace, attr.nodeName, attr.nodeValue ?? '')
+        );
       // el2.firstChild != null
       while (el2.firstChild) {
         el3.appendChild(el2.firstChild);
       }
       el2 = el3;
     }
 
-    for (j = 0, attrs = el.attributes, len = attrs.length; j < len; j++) {
-      attr = attrs.item(j);
+    for (let j = 0, attrs = el.attributes, len = attrs.length; j < len; j++) {
+      const attr = attrs.item(j);
       if (
-        attr.nodeName === 'x' ||
-        attr.nodeName === 'y' ||
-        attr.nodeName === 'xlink:href' ||
-        attr.nodeName === 'href'
+        attr?.nodeName === 'x' ||
+        attr?.nodeName === 'y' ||
+        attr?.nodeName === 'xlink:href' ||
+        attr?.nodeName === 'href'
       ) {
         continue;
       }
 
-      if (attr.nodeName === 'transform') {
+      if (attr?.nodeName === 'transform') {
         currentTrans = attr.nodeValue + ' ' + currentTrans;
-      } else {
-        el2.setAttribute(attr.nodeName, attr.nodeValue);
+      } else if (attr) {
```

Comment:
exactly, we can't wait line 62 to check attr, if we are worried that is false. either we do on top, or we trust is an attribute